sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -327,12 +327,14 @@ class FileContextBasedCheckpointFileManager(path: Path, hadoopConf: Configuratio
   override def renameTempFile(srcPath: Path, dstPath: Path, overwriteIfPossible: Boolean): Unit = {
     import Options.Rename._
     fc.rename(srcPath, dstPath, if (overwriteIfPossible) OVERWRITE else NONE)
+    mayRemoveCrcFile(srcPath)
```

Comment:
The ideal behavior would be renaming crc file to be matched with dstPath, but in SPARK-17475 seems like we just decided to remove the crc file, so I guess this is OK.

Reply:
Is it possible to detect whether checksum is enabled? If it's disabled, we can save an `exists` call.

Reply:
NVM. Just saw your latest comment.

Reply:
Yeah, please feel free to let me know if we prefer alternative approach - I can make a change.